source/jormungandr/jormungandr/realtime_schedule/cleverage.py

```diff
@@ -0,0 +1,147 @@
+# coding=utf-8
+
+# Copyright (c) 2001-2016, Canal TP and/or its affiliates. All rights reserved.
+#
+# This file is part of Navitia,
+#     the software to build cool stuff with public transport.
+#
+# Hope you'll enjoy and contribute to this project,
+#     powered by Canal TP (www.canaltp.fr).
+# Help us simplify mobility and open public transport:
+#     a non ending quest to the responsive locomotion way of traveling!
+#
+# LICENCE: This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+# Stay tuned using
+# twitter @navitia
+# IRC #navitia on freenode
+# https://groups.google.com/d/forum/navitia
+# www.navitia.io
+from jormungandr.realtime_schedule.realtime_proxy import RealtimeProxy
+from flask import logging
+import pybreaker
+import pytz
+import requests as requests
+from jormungandr import cache, app
+from jormungandr.schedule import RealTimePassage
+from datetime import datetime
+
+
+class Cleverage(RealtimeProxy):
+    """
+    class managing calls to cleverage external service providing real-time next passages
+    """
+    def __init__(self, id, service_url, service_args, timezone, object_id_tag=None, timeout=10):
+        self.service_url = service_url if (service_url[-1] == u'/') else (service_url+'/')
+        self.service_args = service_args
+        self.timeout = timeout  # timeout in seconds
+        self.rt_system_id = id
+        self.object_id_tag = object_id_tag if object_id_tag else id
+        self.breaker = pybreaker.CircuitBreaker(fail_max=app.config['CIRCUIT_BREAKER_MAX_CLEVERAGE_FAIL'],
+                                                reset_timeout=app.config['CIRCUIT_BREAKER_CLEVERAGE_TIMEOUT_S'])
+        self.timezone = pytz.timezone(timezone)
+
+    def __repr__(self):
+        """
+         used as the cache key. we use the rt_system_id to share the cache between servers in production
+        """
+        return self.rt_system_id
+
+    @cache.memoize(app.config['CACHE_CONFIGURATION'].get('TIMEOUT_CLEVERAGE', 30))
+    def _call_cleverage(self, url):
+        """
+        http call to cleverage
+        """
+        try:
+            headers = {}
+            for k, v in self.service_args.items() :
+                headers[k] = v
```

Comment:
Why you can't just forward directly self.service_args instead of creating a new dict that is exactly the same ?